agent-inject/agent/agent.go

```diff
@@ -504,7 +513,7 @@ func (a *Agent) Validate() error {
 	}
 
 	if a.ConfigMapName == "" {
-		if a.Vault.Role == "" {
+		if a.Vault.Role == "" && a.Vault.AuthType == DefaultVaultAuthType {
 			return errors.New("no Vault role found")
 		}
 
```

Comment:
It seems like there should be a check here that a.Vault.AuthType isn't blank.

Reply:
Good eye; will add a check before this in a similar fashion to the check for AuthPath in this block

Reply:
Added

Also added a check to allow both the annotations `role` and `auth-config-role` to satisfy the "no Vault role found" check, as they are equivalent